src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -1238,7 +1233,7 @@ private Action<Diagnostic> GetAddDiagnostic(ISymbol contextSymbol, ImmutableArra
         {
             return diagnostic =>
             {
-                if (isGeneratedCodeSymbol && shouldSuppressGeneratedCodeDiagnostic(diagnostic, analyzer, compilation, cancellationToken))
+                if (shouldSuppressGeneratedCodeDiagnostic(diagnostic, analyzer, compilation, cancellationToken))
```

Comment:
Yes, previous check was doing suppression check only for diagnostics reported from generated code symbols. However, analyzer can report diagnostics on generated code symbol while analyzing other user symbols, and we need to suppress those.

Method is cheap for non-configured analyzers or analyzers configured to report diagnostics for generated code. For analyzers not reporting on generated code, it executes [this](http://source.roslyn.io/Microsoft.CodeAnalysis/R/1137a4351ad071ba.html) to identify if diagnostic location is in generated code as per driver's heuristics. We probably need to look at perf traces to tune this code path, now that we call this for more often.